utils/stats.js

```diff
@@ -59,8 +59,62 @@ const stats = async (req, res, authObj) => {
     if(type === 'participants_biospecimen') response = await getTable('participants_biospecimen', isParent, siteCodes);
 
     return res.status(200).json({stats: response, code:200});
-}
+};
+
+const shortNameToTableName = {
+  race: 'participants_race_count_by_sites',
+  age: 'participant_birthYear_by_siteCode',
+  gender: 'participants_sex_count_by_sites',
+  verification: 'participants_verification_status',
+  workflow: 'participants_workflow_status',
+  recruitsCount: 'participants_recruits_count',
+  optOuts: 'participants_optOuts',
+  allModules: 'participants_allModules',
+  moduleOne: 'participants_moduleOne',
+  modulesTwoThree: 'participants_modulesTwoThree',
+  allModulesAllSamples: 'participants_allModulesAllSamples',
+  modulesNone: 'participants_modulesNone',
+  ssn: 'participants_ssn',
+  biospecimen: 'participants_biospecimen',
+};
+
+const shortNameArray = Object.keys(shortNameToTableName);
+
+/**
+ * Retrieve all stats in one call for dashboard display
+ * @param {Request} req 
+ * @param {Response} res 
+ * @param {object} authObj 
+ * @returns 
+ */
+const getStatsForDashboard = async (req, res, authObj) => {
+  if (req.method !== 'GET') {
+    return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
+  }
+
+  const siteCodes = authObj.siteCodes;
+  let data = {};
+  console.log(`Retrieveing stats data for siteCode: ${siteCodes}`);
+
+  try {
+    let promiseArray = [];
+    for (const shortName of shortNameArray) {
+      promiseArray.push(getStatsFromBQ(shortNameToTableName[shortName], siteCodes));
+    }
+
+    const results = await Promise.all(promiseArray);
```

Comment:
👍

Reply:
> Quick question: Does <Promise.all> handle promise rejections ?

Yes. It rejects when any of the promise rejects. Reference [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all).